forms/FormField.php

```diff
@@ -24,153 +30,244 @@ class FormField extends RequestHandler {
+
 	/**
-	 * Name of the template used to render this form field. If not set, then
-	 * will look up the class ancestry for the first matching template where 
-	 * the template name equals the class name.
+	 * Name of the template used to render this form field. If not set, then will look up the class
+	 * ancestry for the first matching template where the template name equals the class name.
 	 *
-	 * To explicitly use a custom template or one named other than the form 
-	 * field see {@link setTemplate()}, {@link setFieldHolderTemplate()}
+	 * To explicitly use a custom template or one named other than the form field see
+	 * {@link setFieldHolderTemplate()}.
 	 *
 	 * @var string
 	 */
-	protected 
-		$template,
-		$fieldHolderTemplate,
-		$smallFieldHolderTemplate;
-		
+	protected $fieldHolderTemplate;
+
+	/**
+	 * @var string
+	 */
+	protected $smallFieldHolderTemplate;
+
 	/**
-	 * @var array All attributes on the form field (not the field holder).
-	 * Partially determined based on other instance properties, please use {@link getAttributes()}.
+	 * All attributes on the form field (not the field holder).
+	 *
+	 * Partially determined based on other instance properties.
+	 *
+	 * @see getAttributes()
+	 *
+	 * @var array
 	 */
 	protected $attributes = array();
 
 	/**
-	 * Takes a fieldname and converts camelcase to spaced
-	 * words. Also resolves combined fieldnames with dot syntax
-	 * to spaced words.
+	 * Takes a field name and converts camelcase to spaced words. Also resolves combined field
+	 * names with dot syntax to spaced words.
 	 *
 	 * Examples:
+	 *
 	 * - 'TotalAmount' will return 'Total Amount'
 	 * - 'Organisation.ZipCode' will return 'Organisation Zip Code'
 	 *
 	 * @param string $fieldName
+	 *
 	 * @return string
 	 */
 	public static function name_to_label($fieldName) {
 		if(strpos($fieldName, '.') !== false) {
 			$parts = explode('.', $fieldName);
-			$label = $parts[count($parts)-2] . ' ' . $parts[count($parts)-1];
+
+			$label = $parts[count($parts) - 2] . ' ' . $parts[count($parts) - 1];
 		} else {
 			$label = $fieldName;
 		}
-		$label = preg_replace("/([a-z]+)([A-Z])/","$1 $2", $label);
-		
-		return $label;
+
+		return preg_replace('/([a-z]+)([A-Z])/', '$1 $2', $label);
 	}
 
 	/**
 	 * Construct and return HTML tag.
+	 *
+	 * @param string $tag
+	 * @param array $attributes
+	 * @param null|string $content
+	 *
+	 * @return string
 	 */
 	public static function create_tag($tag, $attributes, $content = null) {
 		$preparedAttributes = '';
-		foreach($attributes as $k => $v) {
-			// Note: as indicated by the $k == value item here; the decisions over what to include in the attributes
-			// can sometimes get finicky
-			if(!empty($v) || $v === '0' || ($k == 'value' && $v !== null) ) {
-				$preparedAttributes .= " $k=\"" . Convert::raw2att($v) . "\"";
+
+		foreach($attributes as $attributeKey => $attributeValue) {
+			if(!empty($attributeValue) || $attributeValue === '0' || $attributeKey == 'value') {
```

Comment:
Yes but you'll see that if it was null previously and $k was 'value' then it would not be added to the attributes. Now it does. The `!empty()` is on the wrong side of the `||`.